classes/security/UserGroupDAO.inc.php

```diff
@@ -994,6 +1007,42 @@ function getRecommendOnlyGroupIds($contextId, $roleId = null) {
 		return $userGroupIds;
 	}
 
+	/**
+	 * Get all user group IDs with permit_metadata_edit option enabled.
+	 * @param $contextId integer
+	 * @param $roleId integer (optional)
+	 * @return array
+	 */
+	function getPermitMetadataEditGroupIds($contextId, $roleId = null) {
+		$params = array((int) $contextId);
+		if ($roleId) $params[] = (int) $roleId;
+
+		$result = $this->retrieve(
+			'SELECT	ug.user_group_id
+			FROM user_groups ug
+			WHERE permit_metadata_edit = 1 AND
+			ug.context_id = ?
+			' . ($roleId?' AND ug.role_id = ?':''),
+			$params
+		);
+
+		$userGroupIds = array();
+		while (!$result->EOF) {
+			$userGroupIds[] = (int) $result->fields[0];
+			$result->MoveNext();
+		}
+
+		$result->Close();
+		return $userGroupIds;
+	}
+
+	/**
+	 * Get a list of roles not able to change submissionMetadataEdit permission option.
+	 * @return array
+	 */
+	static function getNotChangeMetadataEditPermissionRoles() {
+		return array(ROLE_ID_MANAGER);
```

Comment:
I'm confused -- shouldn't the editor/manager role be able to change this flag?

Reply:
...or does this mean that others cannot disable the flag for editors/managers?

Reply:
Yes that is the idea - that in fact no one can change this setting for the managers - they should always be able to make changes.